src/core.c/CompUnit/Repository/Distribution.pm6

```diff
@@ -30,6 +30,10 @@ class CompUnit::Repository::Distribution does Distribution {
     method id(--> Str:D) { nqp::sha1(self.Str) }
     method meta(CompUnit::Repository::Distribution:D:) { %!meta.item }
 
+    method modified(--> Instant:D) {
+        $!dist.?modified // BEGIN Instant.from-posix(0)
```

Comment:
This is another reason, why a file modification timestamp is not the right tool for the job. Not all Distributions are file based. Everything could be stored in memory or one archive, e.g. when someone implements fatpacking of Raku applications. On the other hand, you will always have access to the meta data itself, which is actually what you really care about.

Reply:
> Not all Distributions are file based.

See my discussion with Nick about that. While, in general, I agree that mtime is not optimal for the job, yet there will always be a way to know when a distribution has been changed.

Speaking of fatpacks and alike, they are rather immutable things where distribution wouldn't change. For that epoch would work perfectly good.